docs/reference/analysis/tokenfilters/keyword-repeat-tokenfilter.asciidoc

```diff
@@ -1,90 +1,164 @@
 [[analysis-keyword-repeat-tokenfilter]]
-=== Keyword Repeat Token Filter
+=== Keyword repeat token filter
+++++
+<titleabbrev>Keyword repeat</titleabbrev>
+++++
 
-The `keyword_repeat` token filter Emits each incoming token twice once
-as keyword and once as a non-keyword to allow an unstemmed version of a
-term to be indexed side by side with the stemmed version of the term.
-Given the nature of this filter each token that isn't transformed by a
-subsequent stemmer will be indexed twice. Therefore, consider adding a
-`unique` filter with `only_on_same_position` set to `true` to drop
-unnecessary duplicates.
+Outputs each token in a stream twice:
+
+* Once as a <<stemmer-keyword,keyword>>, which is excluded from
+  <<stemming,stemming>>
+* Once as a non-keyword, which can be stemmed normally
+
+This allows you to preserve both a stemmed and unstemmed version of each token.
+
+The `keyword_repeat` filter uses Lucene's
+https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/miscellaneous/KeywordRepeatFilter.html[KeywordRepeatFilter].
+
+[TIP]
+====
+If you only want to exclude specific tokens from stemming, we recommend using
+the <<analysis-keyword-marker-tokenfilter,`keyword_marker`>> token filter. 
+====
+
+[[analysis-keyword-repeat-tokenfilter-analyze-ex]]
+==== Example
+
+The following <<indices-analyze,analyze API>> request uses the `keyword_repeat`
+filter to produce both stemmed and unstemmed versions of 
+`quick foxes jumping`:
 
```

Comment:
Same as above...I think these examples that show how you can analyze arbitrary text on the fly to see what happens need a little more context, as it's not how you would ordinarily configure/use an analyzer.

Reply:
I want the emphasis here to be on the input/output of the token filters, not how to set up a customer analyzer.

As the keyword filters both work to block stemming, their examples _do_ include another stemming filter, but otherwise are designed to be as minimal as possible. I don't believe the current text presents them as analyzers you can use off the shelf.